core/blockchain_test.go

```diff
@@ -4326,3 +4332,75 @@ func TestEIP3651(t *testing.T) {
 		t.Fatalf("sender balance incorrect: expected %d, got %d", expected, actual)
 	}
 }
+
+func TestRewindOnConfigChange(t *testing.T) {
+	genesisTime := uint64(12)
+
+	type testCase struct {
+		name              string
+		override1         func(*params.ChainConfig)
+		override2         func(*params.ChainConfig)
+		expectChainRewind bool
+	}
+
+	tcs := []testCase{
+		{
+			name:              fmt.Sprintf("CanyonTime changes from 10 to 0 (genesis time is %d)", genesisTime),
```

Comment:
nit: if using this string as the test name in `t.Run`, I suggest to use a short identifier like just `canyon` here, because this whole string gets converted into the test name. In a failing scenario this would be enough to track down the failing test.